src/main/java/fr/minuskube/inv/content/InventoryProvider.java

```diff
@@ -17,10 +17,18 @@
 package fr.minuskube.inv.content;
 
 import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
 
+/**
+ * TODO: Write well javadoc.
+ *
+ * @since 1.0
+ */
 public interface InventoryProvider {
 
-    void init(Player player, InventoryContents contents);
-    void update(Player player, InventoryContents contents);
+    void init(@NotNull Player player, @NotNull InventoryContents contents);
+
+    default void update(@NotNull final Player player, @NotNull final InventoryContents contents) {
```

Comment:
I like this, it prevents the empty update method from floating around. (But it still technically isn't Clean-Up)